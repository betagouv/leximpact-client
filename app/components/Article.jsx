import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import MuiExpansionPanel from '@material-ui/core/ExpansionPanel';
import MuiExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import MuiExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Fab from '@material-ui/core/Fab';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import AddIcon from '@material-ui/icons/Add';
import SelectControl from "../components/SelectControl"

const ExpansionPanel = withStyles({
  root: {
    border: '0px solid rgba(0,0,0,.125)',
    boxShadow: 'none',
    '&:not(:last-child)': {
      borderBottom: 0,
    },
    '&:before': {
      display: 'none',
    },
  },
  expanded: {
    margin: 'auto',
    padding: '1px',
  },
})(MuiExpansionPanel);

const ExpansionPanelSummary = withStyles({
  root: {
    backgroundColor: 'rgba(0,0,0,0)',
    borderBottom: '0px solid rgba(0,0,0,.125)',
    marginBottom: -1,
    minHeight: 32,
    '&$expanded': {
      minHeight: 20,
    },
  },
  content: {
    '&$expanded': {
      margin: '3px 0',
      padding: '1px',
    },
  },
  expanded: {},
})(props => <MuiExpansionPanelSummary {...props} />);

ExpansionPanelSummary.muiName = 'ExpansionPanelSummary';

const ExpansionPanelDetails = withStyles(theme => ({
  root: {
    padding: theme.spacing.unit * 2,
  },
}))(MuiExpansionPanelDetails);

class CustomizedExpansionPanel extends React.Component {
  state = {
    expanded: 'null', // état de l'extansion panel null = contenu caché
  };

  handleChange = panel => (event, expanded) => {
    this.setState({
      expanded: expanded ? panel : false,
    });
  };

  render() {
    const { expanded } = this.state;
    var styleExpansionpanel = {
    	padding: '1px',
    }
    return (
      <div>

		      	<Typography variant="h2" color="inherit">
		            Article 197
		        </Typography>

		        <Typography variant="overline" color="inherit">
		            - Code général des impôts
		        </Typography>

		        <SelectControl/>



		        <ExpansionPanel
		          square
		          expanded={expanded === 'panel1'}
		          onChange={this.handleChange('panel1')}>

			          <ExpansionPanelSummary style={styleExpansionpanel} expandIcon={<ExpandMoreIcon />}>
				            <Typography variant='body2' color='inherit'>
				            		I. En ce qui concerne les contribuables ... 
				            </Typography>

			          </ExpansionPanelSummary>

			          <ExpansionPanelDetails style={styleExpansionpanel}>
				            <Typography variant='body2' color='inherit'>
				               visés à l'article 4 B, il est fait application des règles suivantes pour le calcul de l'impôt sur le revenu :
				            </Typography>
			          </ExpansionPanelDetails>

		        </ExpansionPanel>

		        <Typography variant="body2" color="inherit">
		            1. L'impôt est calculé en appliquant à la fraction de chaque part de revenu qui excède 9 964€ le taux de :
		        </Typography>

		        <Typography variant="body2" color="inherit">
		            – 14 % pour la fraction supérieure à 9 964 € et inférieure ou égale à 27 519 € ;
		        </Typography>

		        <Typography variant="body2" color="inherit">
		            – 30 % pour la fraction supérieure à 27 519 €  et inférieure ou égale à 73 779 € ;
		        </Typography>

		        <Typography variant="body2" color="inherit">
		            – 41 % pour la fraction supérieure à 73 779 €  et inférieure ou égale à 156 244 € ;
		        </Typography>

		        <Typography variant="body2" color="inherit">
		            – 45 % pour la fraction supérieure à 156 244 €.
		        </Typography>

		        <Fab size="small" color="primary" aria-label="Add" >
         		 	<AddIcon />
        		</Fab>

		        <Button>
            		Ajouter une tranche 
          		</Button>

		 </div>


    );
  }

  }

export default CustomizedExpansionPanel;