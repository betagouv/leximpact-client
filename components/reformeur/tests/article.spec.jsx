/* eslint
    indent: [2, 2],
    semi: [2, "always"],
    react/jsx-indent: [2, 2],
    react/jsx-indent-props: [2, 2]
    max-nested-callbacks: [2, { "max": 4 }]
*/
import { shallow } from "enzyme";

import Article from "../article";
import ReformeDefaultData from "../reforme-default-data";

describe("components | reformeur | article", () => {
  describe("snapshot", () => {
    it("doit correspondre avec les props requises", () => {
      const props = {
        reforme: ReformeDefaultData,
        reformebase: ReformeDefaultData,
        onChange: jest.fn(),
        addTranche: jest.fn(),
        removeTranche: jest.fn(),
      };
      const wrapper = shallow(<Article {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
