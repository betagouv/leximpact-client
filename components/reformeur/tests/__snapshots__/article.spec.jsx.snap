// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | reformeur | article snapshot doit correspondre avec les props requises 1`] = `
<div
  style={
    Object {
      "marginBottom": "1.5em",
      "marginLeft": "1.5em",
      "marginRight": "1.5em",
      "marginTop": "1em",
    }
  }
>
  <WithStyles(WithStyles(ExpansionPanel))
    expanded={false}
    onChange={[Function]}
    square={true}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    <WithStyles(Component)
      expandIcon={<pure(ExpandMoreIcon) />}
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        I. En ce qui concerne les contribuables ...
      </WithStyles(Typography)>
    </WithStyles(Component)>
    <WithStyles(WithStyles(ExpansionPanelDetails))
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        visés à l'article 4 B, il est fait application des règles suivantes pour le calcul de l'impôt sur le revenu :
      </WithStyles(Typography)>
    </WithStyles(WithStyles(ExpansionPanelDetails))>
  </WithStyles(WithStyles(ExpansionPanel))>
  <WithStyles(Typography)
    color="inherit"
    key="0"
    style={
      Object {
        "padding": "5px",
      }
    }
    variant="body2"
  >
    1. L'impôt est calculé en appliquant à la fraction de chaque part de revenu qui excède
    <OutputField
      style={
        Object {
          "color": "#A6A00C",
          "fontWeight": "bold",
          "lineHeight": "10px",
          "padding": "8px",
          "textDecoration": "underline",
        }
      }
      value={9964}
    />
    <InputField
      name="seuil0"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "2px",
          "marginRight": "2px",
          "width": "70px",
        }
      }
      value={9964}
    />
    € le taux de :
  </WithStyles(Typography)>
  <WithStyles(Typography)
    color="inherit"
    key="1"
    style={
      Object {
        "padding": "5px",
      }
    }
    variant="body2"
  >
    –
     
    <OutputField
      style={
        Object {
          "color": "#A6A00C",
          "fontWeight": "bold",
          "lineHeight": "10px",
          "padding": "8px",
          "textDecoration": "underline",
        }
      }
      value="14"
    />
    <InputField
      name="taux0"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "0px",
          "marginRight": "3px",
          "width": "30px",
        }
      }
      value="14"
    />
    % pour la fraction supérieure à
     
    <OutputField
      style={Object {}}
      value={9964}
    />
    € et inférieure ou égale à
    <OutputField
      style={
        Object {
          "color": "#A6A00C",
          "fontWeight": "bold",
          "lineHeight": "10px",
          "padding": "8px",
          "textDecoration": "underline",
        }
      }
      value={27519}
    />
    <InputField
      name="seuil1"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "2px",
          "marginRight": "2px",
          "width": "70px",
        }
      }
      value={27519}
    />
     
    € ;
  </WithStyles(Typography)>
  <WithStyles(Typography)
    color="inherit"
    key="2"
    style={
      Object {
        "padding": "5px",
      }
    }
    variant="body2"
  >
    –
     
    <OutputField
      style={
        Object {
          "color": "#A6A00C",
          "fontWeight": "bold",
          "lineHeight": "10px",
          "padding": "8px",
          "textDecoration": "underline",
        }
      }
      value="30"
    />
    <InputField
      name="taux1"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "0px",
          "marginRight": "3px",
          "width": "30px",
        }
      }
      value="30"
    />
    % pour la fraction supérieure à
     
    <OutputField
      style={Object {}}
      value={27519}
    />
    € et inférieure ou égale à
    <OutputField
      style={
        Object {
          "color": "#A6A00C",
          "fontWeight": "bold",
          "lineHeight": "10px",
          "padding": "8px",
          "textDecoration": "underline",
        }
      }
      value={73779}
    />
    <InputField
      name="seuil2"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "2px",
          "marginRight": "2px",
          "width": "70px",
        }
      }
      value={73779}
    />
     
    € ;
  </WithStyles(Typography)>
  <WithStyles(Typography)
    color="inherit"
    key="3"
    style={
      Object {
        "padding": "5px",
      }
    }
    variant="body2"
  >
    –
     
    <OutputField
      style={
        Object {
          "color": "#A6A00C",
          "fontWeight": "bold",
          "lineHeight": "10px",
          "padding": "8px",
          "textDecoration": "underline",
        }
      }
      value="41"
    />
    <InputField
      name="taux2"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "0px",
          "marginRight": "3px",
          "width": "30px",
        }
      }
      value="41"
    />
    % pour la fraction supérieure à
     
    <OutputField
      style={Object {}}
      value={73779}
    />
    € et inférieure ou égale à
    <OutputField
      style={
        Object {
          "color": "#A6A00C",
          "fontWeight": "bold",
          "lineHeight": "10px",
          "padding": "8px",
          "textDecoration": "underline",
        }
      }
      value={156244}
    />
    <InputField
      name="seuil3"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "2px",
          "marginRight": "2px",
          "width": "70px",
        }
      }
      value={156244}
    />
     
    € ;
  </WithStyles(Typography)>
  <WithStyles(Typography)
    color="inherit"
    key="4"
    style={
      Object {
        "padding": "5px",
      }
    }
    variant="body2"
  >
    – 
    <OutputField
      style={
        Object {
          "color": "#A6A00C",
          "fontWeight": "bold",
          "lineHeight": "10px",
          "padding": "8px",
          "textDecoration": "underline",
        }
      }
      value="45"
    />
    <InputField
      name="taux3"
      onChange={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "marginLeft": "0px",
          "marginRight": "3px",
          "width": "30px",
        }
      }
      value={45}
    />
    % pour la fraction supérieure à 
    <OutputField
      style={Object {}}
      value={156244}
    />
    €.
  </WithStyles(Typography)>
  <div>
    <WithStyles(Fab)
      color="primary"
      onClick={[Function]}
      size="small"
      style={
        Object {
          "margin": "10px",
          "padding": "3px",
        }
      }
    >
      <pure(AddIcon) />
    </WithStyles(Fab)>
    <WithStyles(Typography)
      color="primary"
      inline={true}
      style={
        Object {
          "margin": "10px",
        }
      }
      variant="overline"
    >
      Ajouter une tranche
    </WithStyles(Typography)>
  </div>
  <div>
    <WithStyles(Fab)
      color="primary"
      onClick={[Function]}
      size="small"
      style={
        Object {
          "margin": "10px",
          "padding": "3px",
        }
      }
    >
      <pure(DeleteIcon) />
    </WithStyles(Fab)>
    <WithStyles(Typography)
      color="primary"
      inline={true}
      style={
        Object {
          "margin": "10px",
        }
      }
      variant="overline"
    >
      Supprimer une tranche
    </WithStyles(Typography)>
  </div>
  <WithStyles(WithStyles(ExpansionPanel))
    expanded={false}
    onChange={[Function]}
    square={true}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    <WithStyles(Component)
      expandIcon={<pure(ExpandMoreIcon) />}
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        2. La réduction d'impôt résultant de l'application du quotient familial ...
      </WithStyles(Typography)>
    </WithStyles(Component)>
    <WithStyles(WithStyles(ExpansionPanelDetails))
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        ... ne peut excéder
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="1551"
        />
        <InputField
          name="plafond_qf.maries_ou_pacses"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="1551"
        />
        € par demi-part ou la moitié de cette somme par quart de part s'ajoutant à une part pour les contribuables célibataires, divorcés, veufs ou soumis à l'imposition distincte prévue au 4 de l'article 6 et à deux parts pour les contribuables mariés soumis à une imposition commune. Toutefois, pour les contribuables célibataires, divorcés, ou soumis à l'imposition distincte prévue au 4 de l'article 6 qui répondent aux conditions fixées au II de l'article 194, la réduction d'impôt correspondant à la part accordée au titre du premier enfant à charge est limitée à
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="3660"
        />
        <InputField
          name="plafond_qf.celib_enf"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="3660"
        />
         
        € Lorsque les contribuables entretiennent uniquement des enfants dont la charge est réputée également partagée entre l'un et l'autre des parents, la réduction d'impôt correspondant à la demi-part accordée au titre de chacun des deux premiers enfants est limitée à la moitié de cette somme. Par dérogation aux dispositions du premier alinéa, la réduction d'impôt résultant de l'application du quotient familial, accordée aux contribuables qui bénéficient des dispositions des a, b et e du 1 de l'article 195, ne peut excéder
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="927"
        />
        <InputField
          name="plafond_qf.celib"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="927"
        />
        € ;
        <br />
        Les contribuables qui bénéficient d'une demi-part au titre des a, b, c, d, d bis, e et f du 1 ainsi que des 2 à 6 de l'article 195 ont droit à une réduction d'impôt égale à
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="1547"
        />
        <InputField
          name="plafond_qf.reduc_postplafond"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="1547"
        />
        pour chacune de ces demi-parts lorsque la réduction de leur cotisation d'impôt est plafonnée en application du premier alinéa. La réduction d'impôt est égale à la moitié de cette somme lorsque la majoration visée au 2 de l'article 195 est de un quart de part. Cette réduction d'impôt ne peut toutefois excéder l'augmentation de la cotisation d'impôt résultant du plafonnement. Les contribuables veufs ayant des enfants à charge qui bénéficient d'une part supplémentaire de quotient familial en application du I de l'article 194 ont droit à une réduction d'impôt égale à
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="1728"
        />
        <InputField
          name="plafond_qf.reduc_postplafond_veuf"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="1728"
        />
        € pour cette part supplémentaire lorsque la réduction de leur cotisation d'impôt est plafonnée en application du premier alinéa du présent 2. Cette réduction d'impôt ne peut toutefois excéder l'augmentation de la cotisation d'impôt résultant du plafonnement.
      </WithStyles(Typography)>
    </WithStyles(WithStyles(ExpansionPanelDetails))>
  </WithStyles(WithStyles(ExpansionPanel))>
  <WithStyles(WithStyles(ExpansionPanel))
    expanded={false}
    onChange={[Function]}
    square={true}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    <WithStyles(Component)
      expandIcon={<pure(ExpandMoreIcon) />}
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        3. Le montant de l'impôt résultant de l'application des dispositions précédentes est réduit de...
      </WithStyles(Typography)>
    </WithStyles(Component)>
    <WithStyles(WithStyles(ExpansionPanelDetails))
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        ...
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="30"
        />
        <InputField
          name="plafond_qf.abat_dom.taux_GuadMarReu"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "0px",
              "marginRight": "3px",
              "width": "30px",
            }
          }
          value="30"
        />
        %, dans la limite de
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="2450"
        />
        <InputField
          name="plafond_qf.abat_dom.plaf_GuadMarReu"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="2450"
        />
        € pour les contribuables domiciliés dans les départements de la Guadeloupe, de la Martinique et de la Réunion ; cette réduction est égale à
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="40"
        />
        <InputField
          name="plafond_qf.abat_dom.taux_GuyMay"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "0px",
              "marginRight": "3px",
              "width": "30px",
            }
          }
          value="40"
        />
        %, dans la limite de
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="4050"
        />
        <InputField
          name="plafond_qf.abat_dom.plaf_GuyMay"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="4050"
        />
        €, pour les contribuables domiciliés dans les départements de la Guyane et de Mayotte ;
      </WithStyles(Typography)>
    </WithStyles(WithStyles(ExpansionPanelDetails))>
  </WithStyles(WithStyles(ExpansionPanel))>
  <WithStyles(WithStyles(ExpansionPanel))
    expanded={false}
    onChange={[Function]}
    square={true}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    <WithStyles(Component)
      expandIcon={<pure(ExpandMoreIcon) />}
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        4. a. Le montant de l'impôt résultant de l'application des dispositions précédentes est diminué, dans...
      </WithStyles(Typography)>
    </WithStyles(Component)>
    <WithStyles(WithStyles(ExpansionPanelDetails))
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        ...la limite de son montant, de la différence entre
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value={1196}
        />
        <InputField
          name="decote.seuil_celib"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value={1196}
        />
        € et les
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="trois quarts"
        />
         
        [
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="75"
        />
        <InputField
          name="decote.taux"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "0px",
              "marginRight": "3px",
              "width": "30px",
            }
          }
          value="75"
        />
        %] de son montant pour les contribuables célibataires, divorcés ou veufs et de la différence entre
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value={1970}
        />
        <InputField
          name="decote.seuil_couple"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value={1970}
        />
         
        € et les
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="trois quarts"
        />
         
        [
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="75"
        />
        <OutputField
          style={
            Object {
              "color": "#00A3FF",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
            }
          }
          value="75"
        />
        %] de son montant pour les contribuables soumis à imposition commune.
      </WithStyles(Typography)>
    </WithStyles(WithStyles(ExpansionPanelDetails))>
  </WithStyles(WithStyles(ExpansionPanel))>
  <WithStyles(WithStyles(ExpansionPanel))
    expanded={false}
    onChange={[Function]}
    square={true}
    style={
      Object {
        "padding": "5px",
      }
    }
  >
    <WithStyles(Component)
      expandIcon={<pure(ExpandMoreIcon) />}
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        b. Le montant de l'impôt résultant du a est réduit dans les conditions prévues...
      </WithStyles(Typography)>
    </WithStyles(Component)>
    <WithStyles(WithStyles(ExpansionPanelDetails))
      style={
        Object {
          "padding": "1px",
        }
      }
    >
      <WithStyles(Typography)
        color="inherit"
        variant="body2"
      >
        ...au sixième alinéa du présent b pour les contribuables dont le montant des revenus du foyer fiscal, au sens du 1° du IV de l'article 1417, est inférieur à
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="21036"
        />
        <InputField
          name="plafond_qf.reduction_ss_condition_revenus.seuil2"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="21036"
        />
         
        €, pour la première part de quotient familial des personnes célibataires, veuves ou divorcées, et à
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="42072"
        />
        <OutputField
          style={
            Object {
              "color": "#00A3FF",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
            }
          }
          value="42072"
        />
        €, pour les deux premières parts de quotient familial des personnes soumises à une imposition commune. Ces seuils sont majorés de
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="3797"
        />
        <InputField
          name="plafond_qf.reduction_ss_condition_revenus.seuil_maj_enf"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="3797"
        />
        € pour chacune des demi-parts suivantes et de la moitié de ce montant pour chacun des quarts de part suivants.
        <br />
        Pour l'application des seuils mentionnés au premier alinéa du présent b, le montant des revenus du foyer fiscal est majoré :
        <br />
        1° Du montant des plus-values, déterminées le cas échéant avant application des abattements pour durée de détention mentionnés au 1 de l'article 150-0 D ou à l'article 150-0 D ter et pour lesquelles il est mis fin au report d'imposition dans les conditions prévues à l'article 150-0 D bis, dans leur rédaction en vigueur jusqu'au 31 décembre 2013 ;
        <br />
        2° Du montant des plus-values, déterminées le cas échéant avant application des abattements pour durée de détention mentionnés aux 1 ter ou 1 quater de l'article 150-0 D ou à l'article 150-0 D ter, et des créances mentionnées aux I et II de l'article 167 bis, pour la seule détermination du premier terme de la différence mentionnée au deuxième alinéa du 1 du II bis du même article 167 bis ;
        <br />
        3° Du montant des plus-values mentionnées au I de l'article 150-0 B ter, déterminées le cas échéant avant application de l'abattement pour durée de détention mentionné aux 1 ter ou 1 quater de l'article 150-0 D, pour la seule détermination du premier terme de la différence mentionné au deuxième alinéa du 2° du a du 2 ter de l'article 200 A pour l'application de la seconde phrase du 3° du même a.
        <br />
        Le taux de la réduction prévue au premier alinéa du présent b est de 20 %. Toutefois, pour les contribuables dont les revenus du foyer fiscal, au sens du 1° du IV de l'article 1417, excèdent
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="18984"
        />
        <InputField
          name="plafond_qf.reduction_ss_condition_revenus.seuil1"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "2px",
              "marginRight": "2px",
              "width": "70px",
            }
          }
          value="18984"
        />
         
        €, pour la première part de quotient familial des personnes célibataires, veuves ou divorcées, ou
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="37968"
        />
        <OutputField
          style={
            Object {
              "color": "#00A3FF",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
            }
          }
          value="37968"
        />
        €, pour les deux premières parts de quotient familial des personnes soumises à une imposition commune, ces seuils étant majorés le cas échéant dans les conditions prévues au même premier alinéa, le taux de la réduction d'impôt est égal à
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="20"
        />
        <InputField
          name="plafond_qf.reduction_ss_condition_revenus.taux"
          onChange={[Function]}
          style={
            Object {
              "fontSize": "20px",
              "marginLeft": "0px",
              "marginRight": "3px",
              "width": "30px",
            }
          }
          value="20"
        />
        % multiplié par le rapport entre :
        <br />
        – au numérateur, la différence entre
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="21036"
        />
        <OutputField
          style={
            Object {
              "color": "#00A3FF",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
            }
          }
          value="21036"
        />
        €, pour les personnes célibataires, veuves ou divorcées, ou
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="42072"
        />
        <OutputField
          style={
            Object {
              "color": "#00A3FF",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
            }
          }
          value="42072"
        />
        €, pour les personnes soumises à une imposition commune, ces seuils étant majorés le cas échéant dans les conditions prévues audit premier alinéa, et le montant des revenus mentionnés au troisième alinéa du présent b, et ;
        <br />
        – au dénominateur,
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="2052"
        />
        <OutputField
          style={
            Object {
              "color": "#00A3FF",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
            }
          }
          value="2052"
        />
        €, pour les personnes célibataires, veuves ou divorcées, ou
         
        <OutputField
          style={
            Object {
              "color": "#A6A00C",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
              "textDecoration": "underline",
            }
          }
          value="4104"
        />
        <OutputField
          style={
            Object {
              "color": "#00A3FF",
              "fontWeight": "bold",
              "lineHeight": "10px",
              "padding": "8px",
            }
          }
          value="4104"
        />
        €, pour les personnes soumises à une imposition commune.
        <br />
        Les montants de revenus mentionnés au présent b sont révisés chaque année dans la même proportion que la limite supérieure de la première tranche du barème de l'impôt sur le revenu. Les montants obtenus sont arrondis, s'il y a lieu, à l'euro supérieur.
      </WithStyles(Typography)>
    </WithStyles(WithStyles(ExpansionPanelDetails))>
  </WithStyles(WithStyles(ExpansionPanel))>
</div>
`;
